src/javascript/ast-value.ts

```diff
@@ -30,12 +30,27 @@ function literalToValue(literal: estree.Literal): LiteralValue {
 /**
  * converts unary to string
  */
-function unaryToValue(unary: estree.UnaryExpression): string {
-  const argValue = expressionToValue(unary.argument);
-  if (argValue === undefined) {
-    return;
+function unaryToValue(unary: estree.UnaryExpression): LiteralValue {
```

Comment:
The semantics of this family of functions is in fact to early evaluate to a literal value, or return undefined if this is not possible. This function is not called directly but instead is called by expressionToValue, which has these semantics.